docker/server/entrypoint.sh

```
@@ -130,39 +141,49 @@ if [ "$WORK_DIR" != "/app" ]; then
+      continue
+    fi
+
+    # Although the sentinel only contains [A-Z_] we still escape it for any regex meta-characters.
+    escaped_sentinel=$(printf '%s\n' "$sentinel" | sed -e 's/\\/\\\\/g' -e 's/[][\/.^$*]/\\&/g')
+
+    # For the replacement value, first escape backslashes, then escape any occurrence of
+    # the chosen delimiter and the '&' (which has special meaning in sed replacements).
+    escaped_value=$(printf '%s\n' "$value" | sed -e 's/\\/\\\\/g' -e "s/[${delimiter}&]/\\\\&/g")
+
+    # Now replace the sentinel with the (properly escaped) value in all files in the working directory.
+    find $WORK_DIR/apps -type f -exec sed -i "s${delimiter}${escaped_sentinel}${delimiter}${escaped_value}${delimiter}g" {} +
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)